### PR TITLE
Rewrite district facility downloads to match state downloads

### PR DESCRIPTION
**Pull request by timcheadle**
**Story cards:**
- [sc-6277](https://app.shortcut.com/simpledotorg/story/6277/merge-the-state-and-district-level-report-code-into-one-service)
- [sc-6275](https://app.shortcut.com/simpledotorg/story/6275/add-a-facility-size-column-to-the-district-report)

## Because

The state data download uses only the `Repository` for data, and we needed to bring the former district downloads up to speed.

## This addresses

This keeps the reports separate (to avoid tons of conditionals in a single file), but makes the data match up. I also added facility size, as requested by the CVHOs.

## Test instructions

1. Go to the test app
2. Click on "Reports", make sure you're looking at a district level report (one level below state)
3. Click the "Download" button top right
4. Click "Monthly facility data"

### CODE REVIEW

File: app/services/monthly_district_data_service.rb [hunk 1]
```diff
@@ -6,13 +6,12 @@ def initialize(region, period)
     @months = period.downto(5).reverse
     regions = region.facility_regions.to_a << region
     @repo = Reports::Repository.new(regions, periods: @months)
-    @dashboard_analytics = DistrictAnalyticsQuery.new(region, :month, 6, period.value, include_current_period: true).call
```
Comment (rsanheim) [inline app/services/monthly_district_data_service.rb:9]: 👍

File: app/services/monthly_district_data_service.rb [hunk 2]
```diff
@@ -23,9 +22,10 @@ def report
 
   private
 
-  def section_labels
+  def section_row
     [
-      Array.new(12, nil),
+      # These just add empty spacer columns
+      Array.new(13, nil),
```
Comment (rsanheim) [inline app/services/monthly_district_data_service.rb:28]: do these numbers just correspond to the number of empty columns? If so, would be worth extracting to a constant I think...`EMPTY_SECTION_ROWS` ?

File: app/services/monthly_district_data_service.rb [hunk 3]
```diff
@@ -23,9 +22,10 @@ def report
...
       "New Registrations",
       Array.new(5, nil),
```
Comment (rsanheim) [inline app/services/monthly_district_data_service.rb:30]: and I guess this is just 5 more empty cols?

File: spec/services/monthly_district_data_service_spec.rb [hunk 4]
```diff
@@ -50,24 +50,25 @@ def find_in_csv(csv_data, row_index, column_name)
...
       expect(find_in_csv(csv, region_row_index, "Patients under care as of #{period.adjusted_period.end.strftime("%e-%b-%Y")}")).to eq("2")
-      new_registrations = csv[region_row_index][12..17]
+      new_registrations = csv[region_row_index][13..18]
       expect(new_registrations).to eq(%w[0 0 2 0 0 0])
-      follow_ups = csv[region_row_index][18..23]
+      follow_ups = csv[region_row_index][19..24]
       expect(follow_ups).to eq(%w[0 0 0 1 0 0])
       expect(find_in_csv(csv, region_row_index, "Patients with BP controlled")).to eq("1")
       expect(find_in_csv(csv, region_row_index, "Patients with BP not controlled")).to eq("0")
       # expect(find_in_csv(csv, region_row_index, "Patients with a missed visit")).to eq("1")
```
Comment (rsanheim) [inline spec/services/monthly_district_data_service_spec.rb:68]: remove this commented line?

**Review by rsanheim — COMMENTED** [review]
Overall looks good - some line level comments

File: app/services/monthly_district_data_service.rb (hunk 2 above)
Comment (timcheadle) [reply app/services/monthly_district_data_service.rb]: I cleaned this up with a different approach entirely

File: app/services/monthly_district_data_service.rb [hunk 5]
```diff
@@ -23,113 +22,142 @@ def report
...
+      # These just add empty spacer columns
+      Array.new(region_headers.size + summary_headers.size, nil),
+      "New Registrations",
+      Array.new(month_headers.size - 1, nil),
+      "Follow-up patients",
+      Array.new(month_headers.size - 1, nil),
+      "Treatment outcomes of patients under care",
+      Array.new(outcome_headers.size - 1, nil),
+      "Drug availability",
+      Array.new(drug_headers.size - 1, nil)
```
Comment (rsanheim) [inline app/services/monthly_district_data_service.rb:81]: definitely prefer this approach

**Review by rsanheim — APPROVED** [review]

**Pull request merged**